src/Microsoft.SqlTools.ServiceLayer/DacFx/DacFxService.cs

```diff
@@ -187,16 +187,19 @@ public async Task HandleGenerateDeployPlanRequest(GenerateDeployPlanParams param
                     out connInfo);
             if (connInfo != null)
             {
-                GenerateDeployPlanOperation operation = new GenerateDeployPlanOperation(parameters, connInfo);
-                operation.Execute(parameters.TaskExecutionMode);
-
-                await requestContext.SendResult(new GenerateDeployPlanRequestResult()
+                await BaseService.RunWithErrorHandling(async () =>
```

Comment:
I made those methods `public static` precisely for situations like this, but ideally, `DacFxService` would just extend `BaseService`.  My "eventual" hope is to move more of the service boilerplate into that class in order to clean stuff up.

Reply:
Why is that even a base class? There's nothing it does that requires any actual instance data so you might as well just stick it in a static helper utility class and let people statically import it IMO.